index.js

```diff
@@ -1,30 +1,179 @@
-main(input, 30);
+
+const generateWithRetries = async (inputText, prompt, retries = 0) => {
+    try {
+        return await generate(inputText, prompt);
+    } catch (err) {
+        if (retries < 3) {
+            return await generateWithRetries(inputText, prompt, retries + 1);
+        } else {
+            throw err;
+        }
+    }
+};
+
+const generateFromQueue = async () => {
+    let data;
+    data = await getMessage();
+
+    if (!data.Messages) {
+        console.error('No messages in queue');
+        generateFromQueue();
+        return;
+    }
+
+    let message = data.Messages[0];
+    let messageBody;
+    try {
+        messageBody = JSON.parse(message.Body);
+        if (!isValidMessage(messageBody)) {
+            throw new Error('Invalid message from queue');
```

Comment:
when you throw error here, will it be caught immediately below at line 139 then sent to DLQ directly? when do you want to NOT send to DLQ?

Reply:
Right, I have actually decided to remove the `sendToDLQ` logic altogether because I think it adds too much complexity and is sort of a premature optimization. It might be worth doing if many messages were coming in malformed but really that would be a problem to address at the source